### PR TITLE
ISPN-6893 Scala removal from server/core module

### PR DESCRIPTION
**Pull request by wburns**

### CODE REVIEW

**Comment by slaskawi** [comment]
The code looks really good... awesome amount of work!

Just a heads up - it will collide with Muti-tenancy PR: https://github.com/infinispan/infinispan/pull/4348 but hopefully not that much.

Also those test failures seem to be related.

File: server/core/src/main/java/org/infinispan/server/core/transport/NettyChannelInitializer.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  private JdkSslContext createSslContext(SSLContext sslContext, ClientAuth clientAuth) {
+    //Unfortunately we need to grap a list of available ciphers from the engine.
+    //If we won't, JdkSslContext will use common ciphers from DEFAULT and SUPPORTED, which gives us 5 out of ~50 available ciphers
+    //Of course, we don't need to any specific engine configuration here... just a list of ciphers
+    String[] ciphers = SslContextFactory.getEngine(sslContext, false, false).getSupportedCipherSuites();
+    return new JdkSslContext(sslContext, false, Arrays.asList(ciphers), IdentityCipherSuiteFilter.INSTANCE, null, clientAuth);
+  }
+
+   private ClientAuth requireClientAuth(SslConfiguration sslConfig) {
+      return sslConfig.enabled() ? ClientAuth.REQUIRE : ClientAuth.NONE;
```
Comment (slaskawi) [inline server/core/src/main/java/org/infinispan/server/core/transport/NettyChannelInitializer.java]: This should be `sslConfig.requireClientAuth()` ;)

TL;DR
After turning on [SSL Debug](https://blogs.oracle.com/java-platform-group/entry/diagnosing_tls_ssl_and_https) it turned out that server challenges client for certificate (which was not associated with the client keystore). This suggested that either client cert file is wrong or server challenge is wrong. I was lucky with the latter :)

Comment (wburns) [inline server/core/src/main/java/org/infinispan/server/core/transport/NettyChannelInitializer.java]: Aha, thanks :) - I figured it was in NettyChannelInitializer somewhere, not sure how I didn't see this one.  Thanks!

**Comment by slaskawi** [comment]
You could probably squash this one with https://github.com/infinispan/infinispan/pull/4468. After that you will probably win "the biggest commit" contest ;) Even though there is a strong competitor: https://github.com/infinispan/infinispan/pull/4348

**Comment by wburns** [comment]
I am trying to leave the commits separated by module so it is easier to track the changes.  I will be adding JIRAs for them today.

**Comment by wburns** [comment]
Ready for review/integration.

File: server/core/src/main/java/org/infinispan/server/core/AbstractProtocolServer.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+ * A common protocol server dealing with common property parameter validation and assignment and transport lifecycle.
+ *
+ * @author Galder Zamarreño
+ * @author wburns
+ * @since 4.1
+ */
+public abstract class AbstractProtocolServer<A extends ProtocolServerConfiguration> extends AbstractCacheIgnoreAware
+        implements ProtocolServer<A> {
+
+   private final JavaLog log = LogFactory.getLog(getClass(), JavaLog.class);
```
Comment (tristantarrant) [inline server/core/src/main/java/org/infinispan/server/core/AbstractProtocolServer.java]: Any reason not to rename "JavaLog" to just "Log" like in all other modules ?

Comment (wburns) [inline server/core/src/main/java/org/infinispan/server/core/AbstractProtocolServer.java]: That was the name before, I can change it.

File: server/core/src/main/java/org/infinispan/server/core/AbstractCacheIgnoreAware.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package org.infinispan.server.core;
+
+import org.infinispan.commons.util.concurrent.ConcurrentHashSet;
+
+import java.util.Set;
+
+/**
+ * @author gustavonalle
```
Comment (slaskawi) [inline server/core/src/main/java/org/infinispan/server/core/AbstractCacheIgnoreAware.java]: Could you please add something meaningful to this javadoc?

File: server/core/src/main/java/org/infinispan/server/core/AbstractProtocolServer.java [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+   protected AbstractProtocolServer(String protocolName) {
+      this.protocolName = protocolName;
+   }
+
+   protected void startInternal(A configuration, EmbeddedCacheManager cacheManager) {
+      this.configuration = configuration;
+      this.cacheManager = cacheManager;
+
+      if (log.isDebugEnabled()) {
+         log.debugf("Starting server with configuration: %s", configuration);
```
Comment (slaskawi) [inline server/core/src/main/java/org/infinispan/server/core/AbstractProtocolServer.java]: if we use `debugf`, do we need to check if debug is enabled?

File: server/core/src/main/java/org/infinispan/server/core/CacheIgnoreAware.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package org.infinispan.server.core;
+
+import org.infinispan.commons.util.concurrent.ConcurrentHashSet;
+
+import java.util.Set;
+
+/**
+ * @author gustavonalle
+ * @author wburns
```
Comment (slaskawi) [inline server/core/src/main/java/org/infinispan/server/core/CacheIgnoreAware.java]: Could you please add something meaningful to this javadoc?

File: server/core/src/main/java/org/infinispan/server/core/transport/ExtendedByteBuf.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.util.CharsetUtil;
+import org.infinispan.commons.io.SignedNumeric;
+
+import java.util.Optional;
+
+
+public class ExtendedByteBuf {
```
Comment (slaskawi) [inline server/core/src/main/java/org/infinispan/server/core/transport/ExtendedByteBuf.java]: Could you please add javadocs here?

**Comment by slaskawi** [comment]
Awesome work! Some small, really picky stuff found.

**Comment by galderz** [comment]
Looks good! Bye bye scala 😢 😉 

Needs rebasing though.

**Comment by wburns** [comment]
Rebased, will fix up others.

File: server/core/src/main/java/org/infinispan/server/core/AbstractProtocolServer.java (hunk 4 above)
Comment (wburns) [inline server/core/src/main/java/org/infinispan/server/core/AbstractProtocolServer.java]: This is just how it was before, tbh it probably doesn't really matter since this is only ever invoked once.  The debug enabled is probably just a tiny fraction faster when it isn't enabled.

**Comment by wburns** [comment]
Updated, just wait for a test run to verify nothing is broken before integrating.

**Comment by tristantarrant** [comment]
Pulling !

**Comment by tristantarrant** [comment]
Pushed, thanks
